librz/type/path.c

```diff
@@ -36,74 +36,199 @@ RZ_API void rz_type_path_free(RZ_NULLABLE RzTypePath *tpath) {
 	free(tpath);
 }
 
-static st64 path_walker(const RzTypeDB *typedb, const char *path) {
+/**
+ * \brief False if overflow detected.
+ */
+static inline bool _st64_mult_check_overflow(st64 a, st64 b) {
+	return a == 0 || b >= ST64_MAX / a;
+}
+
+/**
+ * \brief Parses the array subscription 'arr[3]'
+ * \param typedb
+ * \param elem_type Type of array element.
+ * \param path String where struct path is stored.
+ * \param i Index in path pointing to the subscription bracket '['.
+ * \param offset Value to place resulting offset. Set to -1 on error.
+ */
+static RzType *path_walker_parse_bracket(const RzTypeDB *typedb, RZ_NONNULL RzType *elem_type, RZ_NONNULL const char *path, RZ_NONNULL RZ_INOUT size_t *i, RZ_OUT st64 *offset) {
```

Comment:
```suggestion
static RZ_BORROW RzType *path_walker_parse_bracket(const RzTypeDB *typedb, RZ_NONNULL RZ_BORROW RzType *elem_type, RZ_NONNULL const char *path, RZ_NONNULL RZ_INOUT size_t *i, RZ_OUT st64 *offset) {
```

And document the return value.

Reply:
fixed